core/container.go

```diff
@@ -314,6 +314,18 @@ func (mnts ContainerMounts) With(newMnt ContainerMount) ContainerMounts {
 	return mntsCp
 }
 
+func (mnts ContainerMounts) Replace(newMnt ContainerMount) ContainerMounts {
```

Comment:
Probably worth erroring if the `newMnt.Target` wasn't already there (or at least returning an `ok bool`).